src/wash/wash.cpp

```diff
@@ -0,0 +1,132 @@
+#include "wash.hpp"
+
+namespace wash {
+    namespace {
+        uint64_t max_iterations;
+        std::vector<std::string> forces_scalar;
+        std::vector<std::string> forces_vector;
+        std::vector<std::unique_ptr<Kernel>> init_kernels;
+        std::vector<std::unique_ptr<Kernel>> loop_kernels;
```

Comment:
Just wondering why you decided to use unique_ptrs here? Would a `std::vector<Kernel>` work here instead?

Reply:
Apparently you cannot have a vector of polymorphic objects in C++, so I did as suggested by the comment under this answer https://stackoverflow.com/a/16126639

Reply:
Ah good spot. I think its because the std::vector will allocate space only for Kernel objects, which is smaller than its subclasses